packages/legacy-scripting-runner/middleware-factory.js

```diff
@@ -203,18 +203,21 @@ const createBeforeRequest = app => {
 
 const createAfterResponse = app => {
   const authType = _.get(app, 'authentication.type');
+  const autoRefresh =
+    _.get(app, 'legacy.authentication.oauth2Config.autoRefresh') ||
+    _.get(app, 'authentication.oauth2Config.autoRefresh');
 
-  const sessionAuthCheckResponse = (response, z) => {
+  const throwForStaleAuth = (response, z) => {
     if (response.status === 401) {
-      throw new z.errors.RefreshAuthError('Session key needs refreshing');
+      throw new z.errors.RefreshAuthError('Authentication needs refreshing');
     }
     return response;
   };
 
   let afterResponse;
 
-  if (authType === 'session') {
-    afterResponse = sessionAuthCheckResponse;
+  if (authType === 'session' || (authType === 'oauth2' && autoRefresh)) {
+    afterResponse = throwForStaleAuth;
```

Comment:
For some reason we did simulate `throwForStaleAuth` for session auth, but but OAuth2 w/refresh we relied on `core` taking care of it (see comment in `example-app/oauth2.js`). After this PR `core` won't anymore, so to keep the same behavior for legacy-scripting runs we need to simulate it for OAuth2 as well.